test/index.test.ts

```diff
@@ -70,20 +69,11 @@ jest.mock('fs', () => ({
   existsSync: jest.fn(() => true),
   readFileSync: jest.fn(() => 'testpayload'),
 }));
+
+const fromEnvMock = jest.fn();
+
 jest.mock('@aws-sdk/credential-provider-env', () => ({
-  // This is the actual implementation in the SDK ^_^
-  fromEnv: jest.fn().mockImplementation(() => () => {
-    const accessKeyId = process.env['AWS_ACCESS_KEY_ID'];
-    const secretAccessKey = process.env['AWS_SECRET_ACCESS_KEY'];
-    const sessionToken = process.env['AWS_SESSION_TOKEN'];
-    const expiration = process.env['AWS_CREDENTIAL_EXPIRATION'];
-    return {
-      accessKeyId,
-      secretAccessKey,
-      sessionToken,
-      expiration,
-    };
-  }),
```

Comment:
I removed this mockImpl because another mockImpl was given in the before hook that always overrides this.